source/Halibut/Transport/DeflateStreamInputBufferReflector.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.IO.Compression;
+#if !NETFRAMEWORK
+using System.Reflection;
+#endif
+
+namespace Halibut.Transport.Protocol
+{
+
+    #if NETFRAMEWORK
+    /// <summary>
+    /// Not supported on NETFRAMEWORK. Reflects <see cref="DeflateStream"/> to determine the number of bytes available in its input buffer.
+    /// </summary>
+    class DeflateStreamInputBufferReflector
+    {
+        public bool TryGetAvailableInputBufferSize(DeflateStream stream, out uint unusedSizeBytes)
+        {
+            throw new PlatformNotSupportedException();
+        }
+    }
+    #else
+
+    /// <summary>
+    /// Reflects <see cref="DeflateStream"/> to determine the number of bytes available in an instance's input buffer.
+    /// </summary>
+    class DeflateStreamInputBufferReflector
```

Comment:
So this class works on the principal that if the deflate stream reads bytes that weren't originally compressed, internally it is reflected in the number of bytes on a property down in the zLibStream?
Reflection isn't ideal but good to see we have tests around all this 👍 
A comment around this class might be nice to give some background, even if it contains just a link to an issue\PR with further details